zerver/middleware.py

```diff
@@ -306,6 +309,10 @@ def process_response(self, request: HttpRequest,
             client = request.client.name
         except Exception:
             client = request.client_name
+        try:
+            client_version = request.client_version
+        except Exception:
+            client_version = None
```

Comment:
When would we expect an `Exception` to happen here?

Reply:
According to me in any case as the corresponding process_request will always run before the process_response, so we should always have the client_version attribute. But in the circleci
and even on my development server I am getting this error:

```
2020-03-15 18:15:12.563 ERR  [django.request] Internal Server Error: /json/events
Traceback (most recent call last):
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/utils/deprecation.py", line 96, in __call__
    response = self.process_response(request, response)
  File "/home/circleci/zulip/zerver/middleware.py", line 312, in process_response
    client_version = request.client_version
AttributeError: 'WSGIRequest' object has no attribute 'client_version'
```

I couldn't find why the error occured. It would be great if you can help.

Reply:
Ah, I know what the problem is. In our magical `AsyncDjangoHandler` in `zerver/tornado/handlers.py` we do some fiddling with request objects and need to copy over these special attributes. This block:
```
        # Add to this new HttpRequest logging data from the processing of
        # the original request; we will need these for logging.
        #
        # TODO: Design a cleaner way to manage these attributes,
        # perhaps via creating a ZulipHttpRequest class that contains
        # these attributes with a copy method.
        request._log_data = old_request._log_data
        if hasattr(request, "_rate_limit"):
            request._rate_limit = old_request._rate_limit
        if hasattr(request, "_requestor_for_logs"):
            request._requestor_for_logs = old_request._requestor_for_logs
        request.user = old_request.user
        request.client = old_request.client
```

we should copy both the `client_name` and `client_version` from old_request to request in the same way. And then this `except Exception` should be able to be removed.

Reply:
Thanks for debugging. Will get to it.

Reply:
@mateuszmandera fixed it. Removed the try catch in the client_name part too.

Reply:
Thanks, looks good!